dragonfly/implementations/roach2interface.py

```diff
@@ -0,0 +1,316 @@
+
+        cf = self.freq_dict[channel]
+        gain = self.gain_dict[channel]
+
+        logger.info('grabbing {} packets from {}'.format(NPackets*2,dsoc_desc))
+        pkts=ArtooDaq.grab_packets(self, NPackets*2, dsoc_desc, True)
+        logger.info('cf={}'.format(cf))
+        N = 4096
+        p = []
+        t = []
+        for i in range(int(NPackets*2)):
+            if pkts[i].freq_not_time==False:  
+                x=pkts[i].interpret_data()
+                t.append(x)
+                p.append(np.abs(np.fft.fftshift(np.fft.fft(x)))/N)
+        p = np.array(p)
+        t = np.array(t)
+        NPackets = np.shape(p)[0] 
+        if mean == True:
+            p = np.mean(p, axis = 0)
+        if path==None:
+            path=self.monitor_target
+        np.save(path+'/T_packets_ft_channel'+channel+'_cf_'+str(cf)+'Hz_gain_'+str(gain)+'_N_'+str(NPackets), p)
+        np.save(path+'/T_packets_channel'+channel+'_cf_'+str(cf)+'Hz_gain_'+str(gain)+'_N_'+str(NPackets), t)
+
+
+    def get_F_packets(self,dsoc_desc=None, channel='a', NPackets=100, mean=True):
+        if channel=='a':
+            dsoc_desc = (str(self.dest_ip),self.dest_port)
+        elif channel=='b':
+            dsoc_desc = (str(self.dest_ip_b),self.dest_port_b)
+        elif channel=='c':
+            dsoc_desc = (str(self.dest_ip_c),self.dest_port_c)
+
+        cf = self.freq_dict[channel]
+        gain = self.gain_dict[channel]
+
+        logger.info('grabbing packets from {}'.format(dsoc_desc))
+        pkts=ArtooDaq.grab_packets(self, NPackets*2, dsoc_desc, True)
+        logger.info('cf={}'.format(cf))
+        N = 4096
+        p = []
+        for i in range(int(NPackets*2)):
+            if pkts[i].freq_not_time==True:
+                f=pkts[i].interpret_data()
+                p.append(np.abs(f))
+        p = np.array(p)
+        if mean == True:
+            p = np.mean(p, axis = 0)
+        np.save(self.monitor_target+'/F_packets_channel_'+channel+'_cf_'+str(cf)+'Hz_gain_'+str(gain), p)
+
+
+    def get_raw_adc_data(self, count=0, N=1, mean = True):
+        p = []
+        for i in range(N):
+            x = ArtooDaq._snap_per_core(self, zdok=0)
+            x_all = x.flatten('C')
+            logger.info('shape x is {} shape x_all is {}'.format(np.shape(x), np.shape(x_all)))
+            x_all = x_all/128.0*0.25
+            for i in range(16):
+                p.append(np.abs(np.fft.fftshift(np.fft.fft(x_all[i*16384:(i+1)*16384])))/16384)
+        p = np.array(p)
+        if mean == True:
+            p = np.mean(p, axis = 0)
+        logger.info('shape of saved array is: {}'.format(np.shape(p)))
+        np.save(self.monitor_target+'/raw_adc'+str(count), p)
+        logger.info('Raw data saved to {}'.format(self.monitor_target+'/raw_adc'+str(count)+'.npy'))
+
+
+    def calibrate_manually(self, gain1=0.0, gain2=0.42, gain3=0.42, gain4=1.55, off1=3.14, off2=-0.39, off3=2.75, off4=-1.18):
```

Comment:
I got these values from André and he found them using the automatic calibration method.
When the roach gets a signal from a signal generator the automatic adc calibration fails so I tried to use this values to remove the 800MHz spur. 
Currently we don't need this method but I think it might be useful one day to have more control over the adc settings.